src/code.c

```diff
@@ -119,6 +119,8 @@ Obj NewFunctionBody(void)
 */
 static Expr * ADDR_EXPR(Expr expr)
 {
+    GAP_ASSERT(STATE(PtrBody) == PTR_BAG(BODY_FUNC(CURR_FUNC())));
```

Comment:
Doesn't this add major overhead to method execution?

Also, I don't really see how this assertion could ever be violated, given that we update `PtrBody` in a special callback, `VarsAfterCollectBags`.

Reply:
took this one out. I mainly added it because I suspected something dodgy was going on around here, and checking everything (I now think it's fine)